ipn/ipnlocal/local.go

```diff
@@ -3788,6 +3793,46 @@ func (b *LocalBackend) setTCPPortsInterceptedFromNetmapAndPrefsLocked(prefs ipn.
 	b.setTCPPortsIntercepted(handlePorts)
 }
 
+// setServeProxyHandlersLocked ensures there is an http proxy handler for each
+// backend specified in serveConfig. It expects serveConfig to be valid and
+// up-to-date, so should be called after reloadServeConfigLocked.
+func (b *LocalBackend) setServeProxyHandlersLocked() {
+	if !b.serveConfig.Valid() {
+		return
+	}
+	config := b.serveConfig.AsStruct()
+	backends := make(map[string]bool)
```

Comment:
You could avoid this allocation in the case of no proxy handlers by using mak.Set below. But I suppose that'll be rare-ish.